api/http/api_inventory.go

```diff
@@ -693,6 +693,10 @@ func (i *inventoryHandlers) InternalDevicesStatusHandler(w rest.ResponseWriter,
 	}
 
 	for _, id := range ids {
+		if status == "rejected" {
```

Comment:
Can you make this such that only one mongodb "DeleteMany" operation is executed for this case?
Otherwise I'm all for this change. :+1:

Reply:
ok, deviceauth is the only one who calls it, and always with a single id. the only place that calls it with multipe ids is migration command, but it was run only once so far.